/*
 * This file was automatically generated by EvoSuite
 * Mon Mar 20 18:50:59 GMT 2017
 */

package org.cs362.dominion;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.HashMap;
import org.cs362.dominion.Card;
import org.cs362.dominion.GameState;
import org.cs362.dominion.Player;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class GameState_ESTest extends GameState_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      GameState gameState0 = new GameState(1163);
      gameState0.getWinner();
      gameState0.initializeGame();
      HashMap<Card, Integer> hashMap0 = gameState0.supply;
      gameState0.supply = hashMap0;
      gameState0.initializeGame();
      gameState0.supply = hashMap0;
      gameState0.getWinner();
      gameState0.getWinner();
      Card card0 = new Card("pG/xEHftH");
      gameState0.getWinner();
      Integer integer0 = new Integer((-1261));
      hashMap0.put(card0, integer0);
      Card card1 = gameState0.getRandCard();
      // Undeclared exception!
      try { 
        card1.getType();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.Card", e);
      }
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      GameState gameState0 = new GameState(2844);
      // Undeclared exception!
      try { 
        gameState0.getSupplyVal((Card) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      GameState gameState0 = new GameState((-242));
      gameState0.supply = null;
      // Undeclared exception!
      try { 
        gameState0.getRandCard();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      GameState gameState0 = new GameState(2013887228);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      GameState gameState0 = new GameState((-446));
      Card card0 = new Card("EJ^+pRIu^}swF%K1");
      // Undeclared exception!
      try { 
        gameState0.gain(card0, (Player) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      GameState gameState0 = new GameState(1423);
      gameState0.initializeGame();
      HashMap<Card, Integer> hashMap0 = gameState0.embargoList;
      // Undeclared exception!
      try { 
        gameState0.isGameOver();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      GameState gameState0 = new GameState((-1950));
      HashMap<Card, Integer> hashMap0 = new HashMap<Card, Integer>();
      Integer integer0 = new Integer((-2056));
      hashMap0.put((Card) null, integer0);
      // Undeclared exception!
      try { 
        gameState0.getWinner();
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 0 >= 0
         //
         verifyException("java.util.Vector", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      int int0 = (-333376442);
      GameState gameState0 = new GameState((-333376442));
      HashMap<Card, Integer> hashMap0 = gameState0.embargoList;
      gameState0.supply = hashMap0;
      gameState0.initializeGame();
      Card card0 = new Card("@Zisz'@yRZ");
      Integer integer0 = new Integer((-333376442));
      Integer integer1 = new Integer(0);
      Integer.getInteger("", integer1);
      hashMap0.put(card0, integer0);
      // Undeclared exception!
      try { 
        gameState0.play();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      int int0 = 1579;
      Integer integer0 = new Integer(1579);
      Card card0 = new Card("I'4-M[");
      Integer.valueOf(2024111417);
      Integer.getInteger(" al($w");
      GameState gameState0 = new GameState(3778);
      gameState0.getWinner();
      // Undeclared exception!
      try { 
        gameState0.getRandCard();
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      GameState gameState0 = new GameState((-2891));
      HashMap<Card, Integer> hashMap0 = new HashMap<Card, Integer>();
      gameState0.supply = hashMap0;
      Card card0 = new Card("~4N _");
      // Undeclared exception!
      try { 
        gameState0.embargo(card0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      GameState gameState0 = new GameState(7);
      // Undeclared exception!
      try { 
        gameState0.isGameOver();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.cs362.dominion.GameState", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      GameState gameState0 = new GameState(4);
      gameState0.getWinner();
      // Undeclared exception!
      gameState0.initializeGame();
  }
}
